pkg/deploy/controller/deployment/controller.go

```diff
@@ -134,7 +135,7 @@ func (c *DeploymentController) Handle(deployment *kapi.ReplicationController) er
 		// Also, it will scale down the failed deployment and scale back up
 		// the last successful completed deployment
 		if deployutil.IsDeploymentCancelled(deployment) {
-			deployerPods, err := c.podClient.getDeployerPodsFor(deployment.Namespace, deployment.Name)
+			deployerPods, err := deployutil.GetDeployerPodsFor(c.podClient, deployment)
```

Comment:
> An alternative in this case would be to make another util function which returns a selector you can feed to List right here. There are other such functions already in the util to construct selectors. I think that's a decent compromise: shared selector creation, use the List client call directly to make it obvious what the dependencies are.
> 
> What do you think?

In this case I'm certainly willing, but I don't find it unreasonable for unit tests to have to know what data they require and provide that data to a mocked client.  The unit tests are tightly tied to the code either way.